components/form-builder/panel/PanelActionsLocked.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import styled from "styled-components";
+import { FancyButton } from "./Button";
+import { useTranslation } from "next-i18next";
+import useTemplateStore from "../store/useTemplateStore";
+import { LockIcon } from "../icons";
+
+const Actions = styled.div`
+  position: relative;
+  margin-top: 5px;
+  display: flex;
+  background-color: #ebebeb;
+  padding-left: 20px;
+  height: 62px;
+  align-items: center;
+`;
+
+const Label = styled.span`
+  display: flex;
+  align-items: center;
+  line-height: 38px;
+  font-size: 16px;
+  margin-right: 3px;
+  margin-left: 3px;
+
+  & svg {
+    display: inline-block;
+    margin-right: 15px;
+  }
+`;
+
+const AddButtonWrapper = styled.div`
+  position: absolute;
+  top: 40px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 999;
+  right: 20px;
+
+  button {
+    font-size: 16px;
+  }
+`;
+
+export const PanelActionsLocked = ({ addElement }: { addElement: boolean }) => {
+  const { add } = useTemplateStore();
+  const { t } = useTranslation("form-builder");
+  return (
+    <Actions className="panel-actions">
+      <Label>
+        <LockIcon /> {t("Locked element")}
+      </Label>
+      {addElement && (
+        <AddButtonWrapper>
+          <FancyButton
+            onClick={() => {
+              add(0);
+            }}
+          >
+            {t("Add element")}
+          </FancyButton>
```

Comment:
Not sure what's going on here, but the "Add element" button on the locked intro will always add a field to the number 2 position once there is at least one element.

You can always use Move up/down buttons to move them around, but it seems odd that it doesn't insert it into first place if you press the very first "Add element" button.

Reply:
Maybe I'll add a prop for the index --- 

1) Only the intro has the add element button for the others I'm passing the button doesn't show ...

```
<RichTextLocked
            addElement={false}
```

Perhaps in the future we'll have another locked panel that has addElement so I'll update

Reply:
Didn't fully see what you were saying but was chatting with Dave and came up with a fix for now.  The add function adds 1 to the index so it was always hitting `2`.


https://github.com/cds-snc/platform-forms-client/pull/1056/commits/fa43e86736bbc562d61086c90397b1bd34361167

We don't have a use case for other indexes at the moment so I'll leave further updates for later if that works for you.